File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        if (nums.length == 1) return 0;
        int l = 1;
        int r = nums[0];
        int count = 1;

        while (r < nums.length - 1) {
            count++;
            while (l < r) {
                if (nums[l] + l < nums[r] + r) {
                    l++;
                } else {
                    r--;
                }
            }
            r = l + nums[l];
            l = l + 1;
        }
        return count;
    }
}

File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int curr{};
        int farthest{};
        int j{};
        int sz = nums.size() - 1;

        for (auto i{0uz}; i < sz; ++i) {
            farthest = std::max(farthest, static_cast<int>(i) + nums[i]);

            if (i == curr) {
                ++j;
                curr = farthest;
            }
        }

        return j;
    }
};